addons/google_calendar/models/google_sync.py

```diff
@@ -71,8 +71,13 @@ def write(self, vals):
         result = super().write(vals)
         for record in self.filtered('need_sync'):
             if record.google_id:
-                record._google_patch(google_service, record.google_id, record._google_values(), timeout=3)
-
+                # Context keys to handle mail updates according to different recurrence update types ('self_only', 'future', 'all_events')
+                send_updates = self.env.context.get('send_updates', True)
+                mass_archive = self.env.context.get('mass_archive', False)
+                # If the update is from a mass_archive, send a cancellation email only for the first recurrent event
+                if mass_archive and record.start == min(e.start for e in self):
```

Comment:
Indeed, good point! I'm applying the changes right now 🙂

Reply:
Hi Leo! I've pushed some changes inserting the calculation outside the for loop. 🙂

Since the 'write' function receives both Events and Recurrences (and only events have the '.start' attribute), I needed to use the "mass_archive" boolean variable to only calculate the minimum start date when it is True (it is True only when receiving events in 'self', thus it won't access non existent 'start' variable in recurrences). This way I could avoid using the "hasattr" and also the "isistance" function.

What do you think about the update?